### PR TITLE
Fix GDAL 3.7 bindings

### PR DESCRIPTION
**Pull request by lnicola**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: src/dataset.rs [hunk 1]
```diff
@@ -405,8 +405,23 @@ impl Dataset {
     /// Flush all write cached data to disk.
     ///
     /// See [`GDALFlushCache`].
-    pub fn flush_cache(&mut self) {
-        unsafe { GDALFlushCache(self.c_dataset) }
+    ///
+    /// Note: on GDAL versions older than 3.7, this function always succeeds.
+    pub fn flush_cache(&mut self) -> Result<()> {
+        #[cfg(any(all(major_ge_3, minor_ge_7)))]
```
Comment (jdroenner) [inline src/dataset.rs:411]: i think we also need to add `major_ge_4`  to the any block. and in the other case we could add `major_lt_3` but i guess we dont support GDAL versions < 3.

Comment (lnicola) [reply src/dataset.rs]: Yeah, added it now, but I don't think we're really consistent with these.

Comment (jdroenner) [reply src/dataset.rs]: i never thought about a GDAL version > 3 when i "invented" this :see_no_evil: . There will be some trouble when the version really switches to 4...  
Maybe we could write a macro or something to handle this better.

**Review by jdroenner — APPROVED** [review]

**Comment by lnicola** [comment]
bors r=jdroenner

**Comment by lnicola** [comment]
Just for the record, these were broken because I generated the 3.7 bindings on GDAL 3.6.3 😅. Thanks @latot for spotting it.

**Pull request merged**